File: apps/api-kotlin/src/main/kotlin/com/couplecalendar/domain/valueobject/InviteCode.kt
package com.couplecalendar.domain.valueobject

import java.time.Instant
import java.time.temporal.ChronoUnit

data class InviteCode private constructor(
    val value: String,
    val expiresAt: Instant
) {
    fun isExpired(): Boolean = Instant.now().isAfter(expiresAt)
    fun isValid(): Boolean = !isExpired()

    companion object {
        private const val CODE_LENGTH = 6
        private const val CHARS = "ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"
        private const val EXPIRY_HOURS = 24L

        fun generate(): InviteCode {
            val code = (1..CODE_LENGTH)
                .map { CHARS.random() }
                .joinToString("")
            val expiresAt = Instant.now().plus(EXPIRY_HOURS, ChronoUnit.HOURS)
            return InviteCode(code, expiresAt)
        }

        fun fromExisting(code: String, expiresAt: Instant): InviteCode =
            InviteCode(code, expiresAt)
    }
}

File: apps/web/src/domain/entities/InviteCode.ts
/**
 * InviteCode domain entity.
 *
 * 커플 초대 코드 생성 결과. 순수 TypeScript 모델.
 * 만료 시각은 ISO 8601 문자열로 보관한다.
 */
class InviteCode {
	readonly code: string;
	readonly expiresAt: string;

	constructor(code: string, expiresAt: string) {
		this.code = code;
		this.expiresAt = expiresAt;
	}
}

export default InviteCode;
